### PR TITLE
feat(console): support icon in text-input

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

Support "icon" in TextInput. Move "outline" effect to parent div from input.

## Linear Issue Reference

LOG-2041

## Testing

截屏2022-03-29 下午12 26 53

截屏2022-03-29 下午12 27 01

截屏2022-03-29 下午12 27 13

截屏2022-03-29 下午12 27 40

### CODE REVIEW

**Comment by wangsijie** [comment]
#460 merged into this branch by accident

File: packages/console/src/components/Search/index.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import TextInput from '../TextInput';
+import * as styles from './index.module.scss';
+
+type Props = {
+  defaultValue?: string;
+  onSearch?: (value: string) => void;
+};
+
+const Search = ({ defaultValue = '', onSearch }: Props) => {
+  const [inputValue, setInputValue] = useState<string>(defaultValue);
```
Comment (demonzoo) [inline packages/console/src/components/Search/index.tsx:15]: If we use a state and set `defaultValue` as the initial state, then it won't change when the defaultValue changes from upperstream.
Is this by design? If not, then we should consider using a `useEffect` to update `inputValue` state

File: packages/console/src/components/Search/index.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+  const handleSearchKeyPress: KeyboardEventHandler<HTMLInputElement> = (event) => {
+    if (event.key === 'Enter') {
+      onSearch?.(inputValue);
+    }
+  };
```
Comment (demonzoo) [inline packages/console/src/components/Search/index.tsx:21]: Add input debounce

File: packages/console/src/components/Search/index.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+  const handleSearchChange: FormEventHandler<HTMLInputElement> = (event) => {
+    setInputValue(event.currentTarget.value);
+  };
+
+  const handleClick = () => {
+    onSearch?.(inputValue);
```
Comment (demonzoo) [inline packages/console/src/components/Search/index.tsx:28]: nit: Should we check if inputValue is empty?

File: packages/console/src/pages/Users/index.tsx [hunk 4]
```diff
@@ -114,12 +122,14 @@ const Users = () => {
           ))}
         </tbody>
       </table>
-      {totalCount && (
-        <Pagination
-          pageCount={Math.ceil(totalCount / pageSize)}
-          pageIndex={pageIndex}
-          onChange={setPageIndex}
-        />
+      {conditional(
```
Comment (demonzoo) [inline packages/console/src/pages/Users/index.tsx:125]: `conditional` is unnecessary here

File: packages/console/src/components/Search/index.tsx (hunk 2 above)
Comment (wangsijie) [reply packages/console/src/components/Search/index.tsx:21]: We only listen to "Enter" key, so I think "debounce" is not necessary here.

File: packages/console/src/pages/Users/index.tsx (hunk 4 above)
Comment (wangsijie) [reply packages/console/src/pages/Users/index.tsx:125]: `totalCount` can be zero, then will render a `0`. And it also can be "undefined", so `totalCount > 0` is not fit too. Then comes to `conditional`.

File: packages/console/src/components/Search/index.tsx (hunk 3 above)
Comment (wangsijie) [reply packages/console/src/components/Search/index.tsx]: updated, PTAL

File: packages/console/src/components/Search/index.tsx (hunk 2 above)
Comment (demonzoo) [reply packages/console/src/components/Search/index.tsx]: Yes, but what if someone repeatly hitting the enter key? Well, I agree it's non-blocking thing. Let's leave it as-is.

File: packages/console/src/pages/Users/index.tsx (hunk 4 above)
Comment (demonzoo) [reply packages/console/src/pages/Users/index.tsx:125]: `!!totalCount &&` or `Boolean(totalCount) &&` ?

Comment (wangsijie) [reply packages/console/src/pages/Users/index.tsx:125]: `conditional` would be more clear? Not sure if is a good practice to use `conditional` in JSX though.

File: packages/console/src/components/Search/index.tsx (hunk 2 above)
Comment (wangsijie) [reply packages/console/src/components/Search/index.tsx]: Good call, I created LOG-2049 to track this edge case.

File: packages/console/src/pages/Users/index.tsx (hunk 4 above)
Comment (wangsijie) [reply packages/console/src/pages/Users/index.tsx]: removed.

**Review by demonzoo — APPROVED** [review]

**Pull request merged**